Tools/PowershellModule/src/Library/Test-ConnectionToAzure.ps1

```diff
@@ -35,38 +32,34 @@ Function Test-ConnectionToAzure
 
     #>
     PARAM(
-        [Parameter(Position=0, Mandatory=$false)] [string] $SubscriptionName,
-        [Parameter(Position=1, Mandatory=$false)] [string] $SubscriptionId
+        [Parameter(Position=0, Mandatory=$false)] [string] $SubscriptionName = "",
+        [Parameter(Position=1, Mandatory=$false)] [string] $SubscriptionId = ""
     )
-    BEGIN
-    {
-        $Result    = $false
-        $AzContext = Get-AzContext
-    }
-    PROCESS
-    {
-        if($AzContext) {
-            Write-Verbose -Message "Connected to Azure"
-            $Result = $true
 
-            if($AzContext.Subscription.Name -ne $SubscriptionName -and $SubscriptionName) {
-                ## If Subscription Name paramter is passed in, and the value doesn't match current connection return $false
-                Write-Verbose -Message "Connection to an unmatched Subscription in Azure. Not connected to $SubscriptionName"
-                $Result = $false
-            }
-            if($AzContext.Subscription.Id -ne $SubscriptionId -and $SubscriptionId) {
-                ## If Subscription Id paramter is passed in, and the value doesn't match current connection return $false
-                Write-Verbose -Message "Connection to an unmatched Subscription in Azure. Not connected to $SubscriptionId"
-                $Result = $false
-            }
+    $Result    = $false
+    $AzContext = Get-AzContext
+
+    if($AzContext) {
+        if($SubscriptionName -and $AzContext.Subscription.Name -ne $SubscriptionName) {
+            ## If Subscription Name paramter is passed in, and the value doesn't match current connection return $false
+            Write-Error "Connection to an unmatched Subscription in Azure. Not connected to $SubscriptionName"
+            $Result = $false
```

Comment:
nit: Not needed since it's already set above. Same for cases below except success case.